vllm/attention/backends/abstract.py

```diff
@@ -105,6 +105,15 @@ class AttentionMetadata:
     # in block 0, and 1st slot in block 1, respectively.
     slot_mapping: torch.Tensor
 
+    # CountsUpdate is used when multi-step is enabled along with
+    # chunked-prefill. The updated counts reflect the Single Step Prompts
+    # turning into Decodes.
+    @dataclass
+    class CountsUpdate:
+        num_prefills: int
+        num_prefill_tokens: int
+        num_decode_tokens: int
```

Comment:
This data structure could be expanded when more sophisticated techniques are used (like inserting prefills at arbitrary steps)

Reply:
@comaniac I removed this data structure. Instead introduced an argument `turn_prefills_into_decode` to the `advance_step` function.